metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableUtil.java

```diff
@@ -80,7 +80,8 @@ public IcebergTableUtil(final ConnectorContext connectorContext) {
                 final Expression filter = (Expression) new PartitionParser(
                     new StringReader(partitionsRequest.getFilter())).filter()
                     .jjtAccept(icebergFilterGenerator, null);
-                return ScanSummary.of(icebergTable.newScan().filter(filter)).build();
+                return ScanSummary.of(icebergTable.newScan().filter(filter))
+                    .limit(connectorContext.getConfig().getIcebergTableSummaryFetchSize()).build();
             }
```

Comment:
Why we add the limit after applying the filter? The filter-based query should return all the results, not the top n results.

Reply:
Keeping it consistent with the logic without filter. Even with the filter, the results can be huge and slow down the request.

Reply:
OK. I will remove the limit.